### PR TITLE
vterrors: assign some categories

### PR DESCRIPTION
**Pull request by sougou**
I did a round of audit in tabletserver for anonymous errors
and assigned some categories to them.

### CODE REVIEW

File: go/vt/tabletserver/query_splitter.go [hunk 1]
```diff
@@ -97,7 +98,7 @@ func (qs *QuerySplitter) validateQuery() error {
 				}
 			}
 		}
```
Comment (erzel) [inline go/vt/tabletserver/query_splitter.go]: This file is not being used anymore and I forgot to remove it; sorry for causing you some extra work. While you're in here, can you please delete it and the associated test.

Comment (erzel) [inline go/vt/tabletserver/query_splitter.go]: I've only reviewed the changes to the SplitQuery related files.

**Comment by michael-berlin** [comment]
Sugu, I think a couple of these changes have `FAILED_PRECONDITION` where the code should be `INTERNAL` or `INVALID_ARGUMENT`?

Remember that we said that `FAILED_PRECONDITION` means that a retry could succeed at a higher level e.g. on a different vttablet. That's probably not the case for some of the classifications?

**Comment by sougou** [comment]
We should do a tech talk about it. Last time we brainstormed, two approaches were proposed:
1. The error code returned is unchanged through the layers and is interpreted differently based on vtgate vs. app.
2. The error returned by vttablet is only for vtgate, and vtgate must recategorize the code for the app (f needed).

We were leaning towards option 2, but I don't think we finalized.

**Comment by michael-berlin** [comment]
Yes, I agree and I'm in favor of 2.

But I think that discussion is independent of the categorization in this file. My main point here is that this change introduces a few cases which would return `FAILED_PRECONDITION` and I suggest that these should return a more severe, non-retryable error e.g. `INVALID_ARGUMENT` or `INTERNAL`.

**Comment by sougou** [comment]
Can you point out the places where you think things should be different?
BTW, I'll most likely just redo this PR from scratch. It may be easier than handling the merge conflicts at this point.

**Comment by michael-berlin** [comment]
> Can you point out the places where you think things should be different?

1. go/vt/tabletserver/query_executor.go:

    `return nil, vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "invalid cache value for sequence %s: %d", qre.plan.TableName, cache)`

1. go/vt/tabletserver/engines/schema/load_table.go:

    `return vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "%s missing from message table: %s", col, ta.Name.String())`

    and similar errors.

1. go/vt/tabletserver/txthrottler/tx_throttler.go:

    `return vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "Transaction throttler already opened")`

**Comment by sougou** [comment]
Agree with 1 & 3. For 2, it's due to user having misconfigured the message table. But this happens during initialization. So, it's an error seen by the tablet manager, and will eventually result in vttablet not starting. I think failed precondition is still the best fit: something that's misconfigured and needs fixing.

**Comment by sougou** [comment]
Closing this (obsolete).

**Pull request closed without merging**